File: dotnet/Trinsic/Storage/IProfileProvider.cs
using System.Threading;
using System.Threading.Tasks;
using Trinsic.Services.Account.V1;

namespace Trinsic;

public interface IProfileProvider
{
    Task<AccountProfile> GetAsync(string name, CancellationToken cancellationToken = default);

    Task SaveAsync(string name, AccountProfile accountProfile, CancellationToken cancellationToken = default);
}

File: java/src/main/java/trinsic/storage/IProfileProvider.java
package trinsic.storage;

import java.io.IOException;
import java.security.NoSuchAlgorithmException;
import trinsic.services.account.v1.AccountProfile;

public interface IProfileProvider {
  AccountProfile get(String name) throws NoSuchAlgorithmException, IOException;

  void save(String name, AccountProfile accountProfile)
      throws NoSuchAlgorithmException, IOException;
}
